blockmanager.go

```diff
@@ -279,6 +297,34 @@ func newBlockManager(cfg *blockManagerCfg) (*blockManager, error) {
 	}
 	bm.filterHeaderTipHash = fh.BlockHash()
 
+	// If sideload is enabled initialize reader and assign to blockmanager.
+	if bm.cfg.sideLoad.Enabled {
+		log.Infof("Side loading enabled")
+
+		reader, err := chaindataloader.NewBlockHeaderReader(&chaindataloader.ReaderConfig{
+
+			SourceType: bm.cfg.sideLoad.SourceType,
+			Path:       bm.cfg.sideLoad.Path,
+		})
+
+		if err == nil {
+			bm.sideLoadBlkHdrReader = reader
+			cfg := &chanutils.BatchWriterConfig[headerfs.BlockHeader]{
+				QueueBufferSize:        chanutils.DefaultQueueSize,
+				MaxBatch:               BlkHeaderWriteBatch,
+				DBWritesTickerDuration: DBTickerBlkHeaderWriteDuration,
+				PutItems:               bm.cfg.BlockHeaders.WriteHeaders,
```

Comment:
Do we actually need to trickle write this in? On start up, we don't even need to connect to the p2p network to start with, and can just verify the entire site of headers, then write it all to disk in one swoop.

Reply:
>  On start up, we don't even need to connect to the p2p network to start with.

We do not need to connect with p2p network for fetching block headers, is that what you mean?

Reply:
I mean we should insert the new headers to disk before we start up any of the p2p logic. This will dramatically simplify the logic here, as then we don't need to worry about re-orgs or anything like that.